ublox-cellular/src/lib.rs

```diff
@@ -24,6 +24,26 @@ pub use services::data::DataService;
 // Re-export atat version in use
 pub use atat;
 
+pub type Instant<const FREQ_HZ: u32> = fugit::TimerInstantU32<FREQ_HZ>;
+
+#[derive(Debug, PartialEq)]
+pub enum ClockError {
+    Infallible,
+}
+
+/// `Clock` provides all timing capabilities that are needed for the library.
+///
+/// Notice that `Clock` trait uses [fugit](https://lib.rs/crates/fugit) crate for `Duration` and `Instant`.
+pub trait Clock<const FREQ_HZ: u32> {
+    fn now(&mut self) -> fugit::TimerInstantU32<FREQ_HZ>;
+
+    fn start<T>(&mut self, count: T) -> Result<(), ClockError>
+    where
+        T: Into<fugit::MillisDurationU32>;
+
+    fn wait(&mut self) -> Result<(), ClockError>;
+}
+
```

Comment:
`fugit` only provides `Duration` and `Instant` and that is it. Therefore here new `Clock` trait is introduced to work with those types.

This is rough sketch what it should do, maybe it can be simplified. Because essentially we just need `now` and something to make  a `delay`.

Reply:
Note - It might be better to use those types more like this: https://github.com/rtic-rs/systick-monotonic/blob/new-trait-and-time-lib/src/lib.rs#L36-L48